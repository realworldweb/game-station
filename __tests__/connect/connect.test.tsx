/*react testing libary*/
import { act } from 'react-dom/test-utils';
import { render, screen, fireEvent } from '@testing-library/react';

/*components*/
import Connect from '@/pages/connect-four';

/*layouts*/
import Layout from '../../layouts/main';

/*utils*/
import { hasWon, computerTurn } from '@/lib/helpers/connect';

describe('connect', () => {
	it('renders Connect unchanged', () => {
		const { container } = render(<Connect />);

		expect(container).toMatchSnapshot();
	});

	describe('board', () => {
		it('renders board', () => {
			render(<Connect />);

			expect(screen.getByTitle('game-board')).toBeInTheDocument();
		});

		it('renders 8 columns', () => {
			render(<Connect />);

			const columns = screen.getAllByTitle('game-column');
			expect(columns).toHaveLength(8);
		});

		it('renders 9 tiles', () => {
			render(<Connect />);

			const tiles = screen.getAllByTitle('game-tile');
			const tilesNo = tiles.length / 8;
			expect(tilesNo).toBe(9);
		});

		it('fills tile with red if player selects red', () => {
			render(<Connect />);
			const button = screen.getByRole('button', {
				name: /1 player/i,
			});

			fireEvent.click(button);

			const nameForm = screen.getByRole('form', {
				name: /player-names/i,
			});

			fireEvent.submit(nameForm);

			const xButton = screen.getByRole('button', { name: /Red/i });

			fireEvent.click(xButton);

			const columns = screen.getAllByTitle('game-column');

			act(() => {
				fireEvent.click(columns[0]);
			});

			const tiles = screen.getAllByTitle('game-tile');

			expect(tiles[0]).toHaveStyle('background-color: red;');
		});

		it('fills tile with red if player selects red', () => {
			render(<Connect />);
			const button = screen.getByRole('button', {
				name: /1 player/i,
			});

			act(() => {
				fireEvent.click(button);
			});

			const nameForm = screen.getByRole('form', {
				name: /player-names/i,
			});

			act(() => {
				fireEvent.submit(nameForm);
			});

			const xButton = screen.getByRole('button', { name: /Yellow/i });

			act(() => {
				fireEvent.click(xButton);
			});

			const columns = screen.getAllByTitle('game-column');

			act(() => {
				fireEvent.click(columns[0]);
			});

			const tiles = screen.getAllByTitle('game-tile');

			expect(tiles[0]).toHaveStyle('background-color: yellow;');
		});
	});
	describe('computer turn', () => {
		it('should add a yellow tile if none exist', () => {
			const board = [
				['red', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
			];

			const { newBoard } = computerTurn(board, 'yellow');
			const updated = newBoard.flatMap((row) => row);

			expect(updated).toContain('yellow');
		});

		it('should add a yellow tile as close as possible to last', () => {
			const board = [
				['red', '', '', '', '', '', '', '', ''],
				['yellow', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
			];

			const { newBoard } = computerTurn(board, 'yellow');

			expect(newBoard[1][1]).toBe('yellow');
		});

		it('should complete a winning col', () => {
			const board = [
				['yellow', 'yellow', 'yellow', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
			];

			const { newBoard } = computerTurn(board, 'yellow');

			expect(newBoard[0][3]).toBe('yellow');
		});

		it('should complete a winning row', () => {
			const board = [
				['yellow', '', '', '', '', '', '', '', ''],
				['yellow', '', '', '', '', '', '', '', ''],
				['yellow', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
			];

			const { newBoard } = computerTurn(board, 'yellow');

			expect(newBoard[3][0]).toBe('yellow');
		});

		it('should block an oppent from winning on col', () => {
			const board = [
				['red', 'red', 'red', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
			];

			const { newBoard } = computerTurn(board, 'yellow');

			expect(newBoard[0][3]).toBe('yellow');
		});

		it('should block an oppent from winning on row', () => {
			const board = [
				['red', '', '', '', '', '', '', '', ''],
				['red', '', '', '', '', '', '', '', ''],
				['red', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
			];

			const { newBoard } = computerTurn(board, 'yellow');

			expect(newBoard[3][0]).toBe('yellow');
		});

		it('should favour a winning row over blocking opponent row', () => {
			const board = [
				['red', '', '', '', '', '', '', '', ''],
				['red', '', '', '', '', '', '', '', ''],
				['red', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['yellow', '', '', '', '', '', '', '', ''],
				['yellow', '', '', '', '', '', '', '', ''],
				['yellow', '', '', '', '', '', '', '', ''],
			];

			const { newBoard } = computerTurn(board, 'yellow');

			expect(newBoard[3][0]).toBe('');
			expect(newBoard[4][0]).toBe('yellow');
		});

		it('should favour a winning col over blocking opponent col', () => {
			const board = [
				['red', 'red', 'red', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['yellow', 'yellow', 'yellow', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
			];

			const { newBoard } = computerTurn(board, 'yellow');

			expect(newBoard[5][3]).toBe('yellow');
		});
	});
	describe('has won', () => {
		it('should return false if no win', () => {
			const board = [
				['red', '', '', '', '', '', '', '', ''],
				['yellow', '', '', '', '', '', '', '', ''],
				['red', '', '', '', '', '', '', '', ''],
				['red', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
			];
			const check = hasWon(board, 0);

			expect(check).toBe(false);
		});

		it('should return true for winning red column', () => {
			const board = [
				['red', 'red', 'red', 'red', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
				['', '', '', '', '', '', '', '', ''],
			];
			const check = hasWon(board, 0);

			expect(check).toBe(true);
		});
	});

	it('should return true for winning yellow column', () => {
		const board = [
			['yellow', 'yellow', 'yellow', 'yellow', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
		];
		const check = hasWon(board, 0);

		expect(check).toBe(true);
	});

	it('should return true for winning red row', () => {
		const board = [
			['red', '', '', '', '', '', '', '', ''],
			['red', '', '', '', '', '', '', '', ''],
			['red', '', '', '', '', '', '', '', ''],
			['red', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
		];
		const check = hasWon(board, 0);

		expect(check).toBe(true);
	});

	it('should return true for winning yellow row', () => {
		const board = [
			['yellow', '', '', '', '', '', '', '', ''],
			['yellow', '', '', '', '', '', '', '', ''],
			['yellow', '', '', '', '', '', '', '', ''],
			['yellow', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
		];
		const check = hasWon(board, 0);

		expect(check).toBe(true);
	});

	it('should return true for winning yellow diagonal', () => {
		const board = [
			['yellow', '', '', '', '', '', '', '', ''],
			['', 'yellow', '', '', '', '', '', '', ''],
			['', '', 'yellow', '', '', '', '', '', ''],
			['', '', '', 'yellow', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
		];
		const check = hasWon(board, 0);

		expect(check).toBe(true);
	});

	it('should return true for winning red diagonal', () => {
		const board = [
			['red', '', '', '', '', '', '', '', ''],
			['', 'red', '', '', '', '', '', '', ''],
			['', '', 'red', '', '', '', '', '', ''],
			['', '', '', 'red', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
		];
		const check = hasWon(board, 0);

		expect(check).toBe(true);
	});

	it('should return true for winning red diagonal', () => {
		const board = [
			['', '', '', '', '', '', '', 'red', ''],
			['', '', '', '', '', '', 'red', '', ''],
			['', '', '', '', '', 'red', '', '', ''],
			['', '', '', '', 'red', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
			['', '', '', '', '', '', '', '', ''],
		];
		const check = hasWon(board, 0);

		expect(check).toBe(true);
	});
});
