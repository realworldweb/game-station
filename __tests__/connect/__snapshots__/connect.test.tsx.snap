// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connect renders Connect unchanged 1`] = `
<div>
  <div
    class="relative flex flex-wrap content-center justify-center flex-none grow w-100 h-14"
  >
    <div
      class="absolute flex flex-col gap-2 bg-white/75 flex-wrap z-10 w-full h-full justify-center content-center "
      title="signs-selection"
    >
      <button
        class="bg-slate-900 text-white px-4 py-2 rounded-md"
      >
        1 player
      </button>
      <button
        class="bg-slate-900 text-white px-4 py-2 rounded-md"
      >
        2 player
      </button>
    </div>
    <div
      class="relative flex flex-col content-center justify-center w-100"
    >
      <p
        class="mb-5 text-center text-white"
        title="current-player"
      >
        Player: player2
      </p>
      <div
        class="relative flex flex-col mx-auto rotate-270 w-96 h-96"
        title="game-board"
      >
        <div
          class="relative flex justify-center h-12 grow"
          title="game-column"
        >
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="0"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="1"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="2"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="3"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="4"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="5"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="6"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="7"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="8"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
        </div>
        <div
          class="relative flex justify-center h-12 grow"
          title="game-column"
        >
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="0"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="1"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="2"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="3"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="4"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="5"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="6"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="7"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="8"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
        </div>
        <div
          class="relative flex justify-center h-12 grow"
          title="game-column"
        >
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="0"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="1"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="2"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="3"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="4"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="5"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="6"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="7"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="8"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
        </div>
        <div
          class="relative flex justify-center h-12 grow"
          title="game-column"
        >
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="0"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="1"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="2"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="3"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="4"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="5"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="6"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="7"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="8"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
        </div>
        <div
          class="relative flex justify-center h-12 grow"
          title="game-column"
        >
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="0"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="1"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="2"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="3"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="4"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="5"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="6"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="7"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="8"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
        </div>
        <div
          class="relative flex justify-center h-12 grow"
          title="game-column"
        >
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="0"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="1"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="2"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="3"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="4"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="5"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="6"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="7"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="8"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
        </div>
        <div
          class="relative flex justify-center h-12 grow"
          title="game-column"
        >
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="0"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="1"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="2"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="3"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="4"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="5"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="6"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="7"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="8"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
        </div>
        <div
          class="relative flex justify-center h-12 grow"
          title="game-column"
        >
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="0"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="1"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="2"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="3"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="4"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="5"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="6"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="7"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
          <div
            class="relative flex items-center justify-center w-full h-full overflow-hidden bg-blue-700 border-2 border-blue-700"
            id="8"
            title="game-tile"
          >
            <div
              class="relative w-2/3 h-2/3 rounded-full tileCap"
              style="background-color: white;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
